### PR TITLE
Add note on training shared models

### PR DESCRIPTION
**Pull request by fmassa**
Usually when training models with shared parameters, by reflex we share both the `parameters` and the `gradParameters` (if not, `optim` will complain about size mismatches).
But sharing the `gradParameters` while using an optimization procedure that iterates over the modules (like `updateParameters`, or even Facebook's `nn.Optim' I think) is wrong. See for example https://github.com/torch/nn/issues/395

Adding it to the docs to make it clearer.

### CODE REVIEW

File: doc/overview.md [hunk 1]
```diff
@@ -141,3 +141,61 @@ end
...
+-- our optimization procedure will use all the parameters at once, because
+-- it requires the flattened parameters and gradParameters Tensors. Thus,
+-- we need to share both the parameters and the gradParameters
+mlp = nn.Sequential()
+linear = nn.Linear(2,2)
+-- need to share the parameters and the gradParameters as well
+linear_clone = linear:clone('weight','bias','gradWeight','gradBias')
+mlp:add(linear)
+mlp:add(linear_clone)
+parameters, gradParameters = mlp:getParameters()
```
Comment (albanD) [inline doc/overview.md]: Its really minor but wouldn't it be clearer to call these `params` and `gradParams` to be consistent with the names used in the function below?

Comment (fmassa) [inline doc/overview.md]: Yes, I'll change that.
I'll just wait for the others to comment before doing the changes.

Comment (soumith) [inline doc/overview.md]: can be merged after changes requested by Alban

**Comment by fmassa** [comment]
@soumith @albanD just updated with the changes proposed by Alban and squashed the commits.

**Comment by soumith** [comment]
thanks Francisco.

**Pull request merged**